### DISCUSSION TITLE
How to get the first and last item of a relation in a mongodb query?

### DISCUSSION THREAD

**Discussion by arthurfiorette** [prisma/prisma#14457] [Q&A] [answered]
Considering the below schema, how can i query a user and retrieve its first and last post.

```prisma
model User {
  posts Post[]
}

model Post {
  userId String @db.objectId
  user   User   @relation(fields: [userId], references: [id])

  createdAt DateTime @default(now()) @db.Date
}
```

I was looking for something like this:

```ts
const usersWithPosts = await PRISMA.user.findMany({
  where: { id: { in: [...] } },
  select: {
    name: true,
    posts: {
      // first and last
    }
  }
});
```

[This stackoverflow post](https://stackoverflow.com/a/68195629/14681561) shows how to do with aggregations, but i'm not familiar with its interface that prisma provides.

Any alternative that doesn't rely on me doing 2 extra db requests for each user is also welcome.

**Comment by austincrim** [comment, marked as answer]
Hey @arthurfiorette 👋,

As far as I'm aware, it isn't possible to do this in a single Prisma query. It might be possible to do with Prisma's [`aggregateRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#aggregateraw), which takes a `pipeline` of plain MongoDB aggregations. All aggregation options can be found here [in the MongoDB documentation](https://www.mongodb.com/docs/manual/reference/operator/aggregation-pipeline/).

Depending on what you are ordering by, the only alternative I can think of is to use `Promise.all` with the `take` option:
```ts
let [first, last] = await Promise.all([
  prisma.user.findMany({
    select: {
      posts: {
        take: 1
      }
    }
  }),
  prisma.user.findMany({
    select: {
      posts: {
        take: -1
      }
    }
  })
])
```

**Reply by arthurfiorette** [reply]
Should I use `Promise.all` or `Prisma.$transaction`?

**Reply by Timmybankz14** [reply]
Prisma.$transaction is the better option in this case.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
